invader_payment_bank_transfer/services/payment_bank_transfer.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2019 ACSONE SA/NV (http://acsone.eu).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from cerberus import Validator
+from openerp.addons.component.core import AbstractComponent
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentBankTransfer(AbstractComponent):
+
+    _name = "payment.service.bank_transfer"
+    _inherit = "base.rest.service"
+    _usage = "payment_bank_transfer"
+    _description = "REST Services for bank transfer payments"
+
+    def _validator_add_payment(self):
+        schema = {"payment_mode": {"type": "string"}}
+        schema.update(
+            self.component(
+                usage="invader.payment"
+            )._invader_get_target_validator()
+        )
+        return schema
+
+    def _validator_return_add_payment(self):
+        schema = {}
+        return Validator(schema, allow_unknown=True)
+
+    def add_payment(self, target, payment_mode, **params):
+        """ Prepare data for SIPS payment submission """
+        payable = self.component(
+            usage="invader.payment"
+        )._invader_find_payable_from_target(target, **params)
+        payment_mode = self.env["account.payment.mode"].browse(
+            int(payment_mode)
+        )
+        transaction = self.env["payment.transaction"].browse()
+        payable._invader_payment_start(transaction, payment_mode)
+        payable._invader_payment_success(transaction)
```

Comment:
We must work on this. I know `_invader_payment_success` is here to trigger the cart confirmation. Yet it's a bit strange to say the payment is successful. @lmignon

Reply:
Also don't we want to create a `payment.transaction`?

Reply:
@sbidoul I'm just getting familiar w/ this but to me seems reasonable that we don't create a transaction here: there's not transaction yet. Transaction will be added manually via backend as soon as the payment is received.
Am I missing anything?

We could argue that w/ manual payment the cart / order should be in another state, eg: "Waiting payment".

Reply:
About `payment.transaction` there are two reasons to create it IMO:
- standard Odoo does it (it creates a transaction in `pending` state), and we want to be as close as possible to Odoo standard here
- it could become something where we hook some automation when the bank transfer is confirmed; if we don't create the transaction we don't have much trace that a payment operation was intended

Reply:
I see. Then maybe you are right. I'll check.

Reply:
> We could argue that w/ manual payment the cart / order should be in another state, eg: "Waiting payment"

I would not add a new SO state for that. The payment status can be deduced from the transaction statuses.

Reply:
So following #4 you can now call `_invader_payment_accepted` instead of `_invader_payment_accepted`. See also some additional thoughts in the docstrings in #4.